src/main/java/shop/dao/impl/UserDAOImpl.java

```diff
@@ -0,0 +1,38 @@
+package shop.dao.impl;
+
+import shop.dao.UserDAO;
+import shop.entity.User;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class UserDAOImpl implements UserDAO {
+    private Map<String, User> userMap = new HashMap<>();
+
+    @Override
+    public void saveUser(User user) {
+        userMap.put(user.getId(), user);
+    }
+
+    @Override
+    public void updateUser() {
+    }
+
+    @Override
+    public void deleteUser() {
+
+    }
+
+    @Override
+    public boolean isExist(String email) {
+        return getUser(email).isPresent();
+    }
+
+    @Override
+    public Optional<User> getUser(String email) {
```

Comment:
getUser -> findUser
it is important because 'get' should return user or throw an exception, find - return user if present